arroyo-sql/src/lib.rs

```diff
@@ -446,6 +446,15 @@ impl<'a> SqlProgramBuilder<'a> {
             let name = name.to_string();
             let mut with_map = HashMap::new();
             for option in with_options {
+                if !matches!(
+                    option.name.value.to_lowercase().as_str(),
+                    "connection" | "topic" | "format" | "event_time_field" | "watermark_field"
```

Comment:
This will break HTTP sources. Generally I don't think it makes sense to have a fixed list here, since every connector will have its own options.

Reply:
I adjusted this to be source-specific. Still feels like a good idea to have validation on this.